volatility3/framework/plugins/linux/pidhashtable.py

```diff
@@ -44,17 +44,16 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
             ),
         ]
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.vmlinux = None
-        self.vmlinux_layer = None
-
     def _is_valid_task(self, task) -> bool:
-        return bool(task and task.pid > 0 and self.vmlinux_layer.is_valid(task.parent))
+        vmlinux = self.context.modules[self.config["kernel"]]
+        vmlinux_layer = self.context.layers[vmlinux.layer_name]
+        return bool(task and task.pid > 0 and vmlinux_layer.is_valid(task.parent))
 
     def _get_pidtype_pid(self):
```

Comment:
I have to admit, I thought these would end up being parameterized classmethods, but as long as you're happy that they're only for use in this plugin this is fine.